### ISSUE TITLE
Some bugs in the program "DESeq2Constructor.R"

### ISSUE THREAD

**Issue by HaiHuaLiu** [williamritchie/IRFinder#16]
Hi Dadi,

There are some bugs in "DESeq2Constructor.R",.
1. On the 22th line, the variable "files" is never defined before, I think it should be "filePaths".   Otherwise, the error will occur "Error in as.vector(files$V1) : object 'files' not found".

2. I think you need to add column names and row names for the variable "res.rd", colnames(res.rd)=as.vector(designMatrix[,1])
rownames(res.rd)=irnames
Otherwise, it cannot run successfully. 

I don't know if I am right, please check it. Look forward to your reply.

Haihua.

**Comment by dg520** [comment]
Hi Haihua,

You're right and I've also realized the bug there. I'll fix this during weekend. Thanks a lot.

Best,
Dadi

**Comment by dg520** [comment]
Hi Haihua,

A new version just released. The bug is fixed. And more importantly, I launched a new GLM approach in this release, which is believed to improve the robustness of modeling. You might want to be noticed about this.

Best,
Dadi

**Comment by dimialex** [comment]
Hi Dadi,
using the same colnames on res.rd and libsz.rd is causing an error later on when trying to create the DESeqDataSetFromMatrix:
```
dd = DESeqDataSetFromMatrix(countData = counts.IRFinder, colData = group, design = ~1)
Error in `rownames<-`(`*tmp*`, value = colnames(countData)) : 
 duplicate rownames not allowed
```
I added a suffix on colnames(libsz.rd) and it could run through. So I replaced the following line on DESeq2Constructor.R:
`colnames(libsz.rd)=as.vector(designMatrix[,1])`
with the following:
`colnames(libsz.rd)=paste(as.vector(designMatrix[,1]),"libsz",sep="_")`

Best,
Dimi

**Comment by dg520** [comment]
Hi Dimi,
Which version of DESeq2 are you using? 

I think the error comes from the version difference of DESeq2. I'm running version 1.10.1, the highest version under R 3.2, although the latest version is DESeq2 v1.16.1.

In my version, DESeq2 will NOT rebuild row names of design matrix according to column names of count matrix if they DO NOT match each other. If you look at the example on the wiki page, the row names of the design matrix is numeric `"1" "2" "3" "4"`. That is done in purpose to avoid the duplication error.

You can check `DESeqDataSetFromMatrix` to see how it deals with the names in your DESeq2 version. 

Adding a suffix is a reasonable way to avoid duplication as well. I'll mark this down and consider it in the next release. 

Thanks a lot for reporting the error.

Best,
Dadi

**Comment by dimialex** [comment]
Hi Dadi,

I'm using DESeq2_1.16.1, here is my sessionInfo():
```
R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.0
LAPACK: /usr/lib/lapack/liblapack.so.3.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DESeq2_1.16.1              SummarizedExperiment_1.6.3 DelayedArray_0.2.7         matrixStats_0.52.2        
 [5] Biobase_2.36.2             GenomicRanges_1.28.4       GenomeInfoDb_1.12.2        IRanges_2.10.2            
 [9] S4Vectors_0.14.3           BiocGenerics_0.22.0       

loaded via a namespace (and not attached):
 [1] genefilter_1.58.1       locfit_1.5-9.1          splines_3.4.1           lattice_0.20-35        
 [5] colorspace_1.3-2        htmltools_0.3.6         base64enc_0.1-3         blob_1.1.0             
 [9] survival_2.41-3         XML_3.98-1.9            rlang_0.1.1             DBI_0.7                
[13] foreign_0.8-69          BiocParallel_1.10.1     bit64_0.9-7             RColorBrewer_1.1-2     
[17] GenomeInfoDbData_0.99.0 plyr_1.8.4              stringr_1.2.0           zlibbioc_1.22.0        
[21] munsell_0.4.3           gtable_0.2.0            htmlwidgets_0.9         memoise_1.1.0          
[25] latticeExtra_0.6-28     knitr_1.16              geneplotter_1.54.0      AnnotationDbi_1.38.1   
[29] htmlTable_1.9           Rcpp_0.12.12            acepack_1.4.1           xtable_1.8-2           
[33] scales_0.4.1            backports_1.1.0         checkmate_1.8.3         Hmisc_4.0-3            
[37] annotate_1.54.0         XVector_0.16.0          bit_1.1-12              gridExtra_2.2.1        
[41] ggplot2_2.2.1           digest_0.6.12           stringi_1.1.5           grid_3.4.1             
[45] tools_3.4.1             bitops_1.0-6            magrittr_1.5            RSQLite_2.0            
[49] lazyeval_0.2.0          RCurl_1.95-4.8          tibble_1.3.3            Formula_1.2-2          
[53] cluster_2.0.6           Matrix_1.2-10           data.table_1.10.4       rpart_4.1-11           
[57] nnet_7.3-12             compiler_3.4.1   
```
Best,
Dimi

**Comment by dg520** [comment]
Hi Dimi,

How did you read your experiment design file into R? And could you please share me that file so I can have a test on my side? My email is at the end of Wiki home page. Thanks.

Best,
Dadi

**Comment by dg520** [comment]
@HaiHuaLiu @dimialex I just released version 1.2.2, which is supposed to fix the bug due to the conflict among row or column names. -Dadi

**Comment by HaiHuaLiu** [comment]
Thanks, it works successfully now.
